sys/usb/usbus/hid/hid_io.c

```diff
@@ -73,10 +74,19 @@ static void _hid_rx_pipe(usbus_hid_device_t *hid, uint8_t *data, size_t len)
     for (size_t i = 0; i < len; i++) {
         isrpipe_write_one(&_hid_stdio_isrpipe, data[i]);
     }
+
+    if (_rx_cb) {
+        _rx_cb->cb(_rx_cb->arg);
+    }
 }
 
 void usb_hid_io_init(usbus_t *usbus, uint8_t *report_desc,
                      size_t report_desc_size)
 {
     usbus_hid_init(usbus, &hid, _hid_rx_pipe,  report_desc, report_desc_size);
 }
+
+void usb_hid_io_set_rx_cb(usb_hid_io_t *usb_hid_io)
```

Comment:
Is there an advantage on keeping a reference to this structure, that basically holds the arguments here? Why not have:
```suggestion
void usb_hid_io_set_rx_cb(usb_hid_io_cb_t cb, void *arg)
```

Reply:
> Also I think it is a good preparation for future PR's where `hid_io.c` might have have to manage multiple HID devices at once.

I see, but then it looks more like `usb_hid_iot_t` should be used in `usb_hid_io_init`, and this function would be something like `void usb_hid_io_set_rx_cb(usb_hid_io_t *usb_hid_io, usb_hid_io_cb_t cb, void *arg)`, right? Unless I misunderstood.

Reply:
No. `usb_hid_io_init` is called to initialize the USB layer. I want the callback functionality to be an additional, not necessarily required, functionality. Possibly the naming of the struct is bad as it suggests that this struct is supposed to be used in a more genral context than just for the callback functionality.

Reply:
> I want the callback functionality to be an additional, not necessarily required, functionality

Agreed.

> Possibly the naming of the struct `usb_hid_io_t` is bad as it suggests that this struct is supposed to be used in a more genral context than just for the callback functionality.

Also agreed. But if this module will be a singleton and not allow multiple instantiations, the pattern used by xtimer is not necessary. I don't see the need to force the user to allocate such a structure, when it could be instantiated statically by the module (or some other `usb_hid_..` instance in the future).

Reply:
Once multiple HID reports are supported by the USB HID layer, this module will no longer be a singleton. Therefore I thought it would be a good idea to introduce a structure for this right now.

Reply:
Yeah I see your point. Removed this struct and adjusted it to also simply use a static arg for now